### ISSUE TITLE
Re-import checkmarks

### ISSUE THREAD

**Issue by jondo** [academic-moodle-cooperation/moodle-mod_checkmark#51]
Is there any way to backup and restore checkmarks?
E.g., is there any way to re-import the checkmarks that can be exported via the Checkmark activities' `Export`tab?
Exporting grades does not contain the info which checkmarks were crossed. Also, course backups do not contain participant-specific data.

**Comment by goggo24** [comment]
If you duplicate a checkmark instance no user data is included. If you backup it with user data, the info should be included after restore. That's the same behavior as for assignment. There is no way to re-import the export information, which you download in the export tab. 

What do you mean with _"Exporting grades does not contain the info which checkmarks were crossed"_? In the export tab you can define which columns should be included (also the individual  crossed checkmarks).  Does this info help? 

PS: Do you use our checkmark report (local/checkmarkreport -  https://github.com/academic-moodle-cooperation/moodle-local_checkmarkreport)  too? It's an extension to the module checkmark and shows the marks and grades for one or more checkmarks in a tabular form and has more export settings than the activity.

**Comment by jondo** [comment]
Thank you for the help!

With _"Exporting grades does not contain the info which checkmarks were crossed"_ I meant exporting via the grade administration. You are right that the individual crossed checkmarks can be exported in the export tab of checkmark assignments or via the checkmark report.

Do I understand correctly that the only way to restore backups of individual crossed checkmarks is via the course backup? In our Moodle instance, when doing a course backup, all user data is blocked (cannot be selected for backup).

**Comment by goggo24** [comment]
Yes you are right - in Gradebook only the points are shown and not the individual checked checkmarks. If you want to restore it in another checkmark instance itfself, you have to use course backup with user data. There is no way to export a file in the "old" checkmark activity and re-import it into a "new" one. 
BUT with v.3.8.0 (just published) as a teacher you can overwrite the students individual crossed checkmarks or if there are not one, a teacher can transfer the crossed checkmarks manually. Overwritten checkmarks are marked for students and teachers, so that the change is traceable - see image. 
Have fun with the new feature!
![overwritten](https://user-images.githubusercontent.com/16721900/74029653-6de33c80-49ad-11ea-8df8-05efb77b8623.png)

**Comment by jondo** [comment]
Thank you for this hint! This is a nice feature in itself. However, in my case it does not serve as workaround for an automated re-import, since I have got too many students for "manual re-import".

This issue is not _that_ urgent, however, because in case of data loss a simple re-import of just the resulting grades will be sufficient.

**Comment by goggo24** [comment]
At the moment we will not pursue the request "reimport" any further. If something changes in our roadmap, I will let you know.
